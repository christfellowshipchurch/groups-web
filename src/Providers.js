import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
