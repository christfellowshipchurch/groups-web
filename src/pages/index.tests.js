import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../Providers';

import Index from '.';

describe('the home page', () => {
  it('should render', () => {
    const tree = renderer
      .create(
        <Providers>
          <Index />
        </Providers>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
