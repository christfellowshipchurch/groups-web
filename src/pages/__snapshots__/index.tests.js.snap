// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the home page should render 1`] = `
Array [
  .c0 {
  font-size: 2.6875rem;
  font-weight: 900;
  line-height: 3.0906249999999997rem;
  margin: initial;
}

<h1
    className="c0 sc-1dj0tjq-2 kEIKSh"
  >
    My page
  </h1>,
  <ul>
    <li>
      <a
        href="/boom"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Boom
      </a>
    </li>
    <li>
      <a
        href="/my-groups"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        My Groups
      </a>
    </li>
  </ul>,
]
`;
